decoder.go

```diff
@@ -119,7 +119,7 @@ func convertUnsignedInts(src interface{}, dst *reflect.Value) error {
 			return nil
 		}
 	case uintptr:
-		if t <= math.MaxUint64 && !dst.OverflowUint(uint64(t)) {
+		if uint64(t) <= math.MaxUint64 && !dst.OverflowUint(uint64(t)) {
```

Comment:
Ditto here.  Casting to uint64 makes the comparison moot.  Until there is a datatype larger than uint64 that uintptr could have and that you could cast both sides to, I don't think this comparison provides any value.

Reply:
Yeah, late night PR's should be prohibited :)